src/sdk/glacier.js

```diff
@@ -271,8 +330,93 @@ class GlacierBackend {
         // removing the request extended attribute. In such case, NSFS would still
         // report the object restore status to be `ONGOING` and we are going
         // to allow a retry of that entry.
-        return restore_status.state === GlacierBackend.RESTORE_STATUS_ONGOING;
+        return restore_status.state === Glacier.RESTORE_STATUS_ONGOING;
+    }
+
+    /**
+     * storage_class_from_xattr returns a parsed storage class derived from the given
+     * extended attribute. It will use DMAPI EA if `use_dmapi` is set to true.
+     *
+     * NOTE: For `use_dmapi` to work, the xattr must have been retrieved using fs_context
+     * where backend is set to 'GPFS'.
+     *
+     * @param {nb.NativeFSXattr} xattr
+     * @param {Boolean} [use_dmapi]
+     * @returns {nb.StorageClass}
+     */
+   static storage_class_from_xattr(xattr, use_dmapi = config.NSFS_GLACIER_USE_DMAPI) {
```

Comment:
Consider `NSFS_GLACIER_USE_DMAPI` to just enable reading DMAPI xattr and have a different config which controls the implicit `GLACIER` storage class.

`config.NSFS_GLACIER_USE_DMAPI` can enable 4 features but the base should be to just read DMAPI xattr:
1. Implicit restore status
2. Implicit GLACIER
3. Tape header info
4. Force Takeover